pkcs11/src/new/types/session.rs

```diff
@@ -5,21 +5,35 @@ use pkcs11_sys::*;
 pub struct Session<'a> {
     handle: CK_SESSION_HANDLE,
     client: &'a Pkcs11,
+    // This is not used but to prevent Session to automatically implement Send and Sync
+    _guard: *mut u32,
 }
 
+// Session does not implement Sync to prevent the same Session instance to be used from multiple
+// threads concurrently because that is unsafe in most cases.
```

Comment:
```suggestion
// Session does not implement Sync to prevent the same Session instance to be used from multiple
// threads.
```
You're completely preventing the `Session` from being sent across threads, and that's not inherently unsafe.

Reply:
Note that I implement `Send` on line 14 to allow `Session` to be sent to other threads, this prevents references of `Session` to be sent to other threads.

It's true that depending on which operations are done concurrently this could be safe, that's why I wrote "in most cases". I will apply your suggestion to be more precise.